Mathlib/Data/Equiv/Functor.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2019 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Simon Hudon, Scott Morrison
+-/
+import Mathlib.Data.Equiv.Basic
+
+/-!
+
+# Functors can be applied to `equiv`s.
+
+```
+def functor.map_equiv (f : Type u → Type v) [functor f] [is_lawful_functor f] :
+  α ≃ β → f α ≃ f β
+```
+
+-/
+
+universes u v w
+
+variable {α β : Type u}
+open equiv
+
+namespace Functor
+
+variable (f : Type u → Type v) [Functor f] [LawfulFunctor f]
+
+-- this is in control.basic in Lean 3
+theorem map_map (m : α → β) (g : β → γ) (x : f α) :
+  g <$> (m <$> x) = (g ∘ m) <$> x :=
+(comp_map _ _ _).symm
+
+/-- Apply a functor to an `equiv`. -/
+def map_equiv (h : α ≃ β) : f α ≃ f β :=
+{ to_fun    := map h,
+  inv_fun   := map h.symm,
+  left_inv  := λ x => by simp [map_map]
+  right_inv := λ x => by simp [map_map] }
```

Comment:
```suggestion
def map_equiv (h : α ≃ β) : f α ≃ f β where
  to_fun    := map h
  inv_fun   := map h.symm
  left_inv x := by simp [map_map]
  right_inv x := by simp [map_map]
```
You don't have to change it, but I just wanted to point out a neat stylistic possibility.

Reply:
Yes I am beginning to understand now that this is for some reason the preferred way.